src/briefcase/integrations/download.py

```diff
@@ -97,3 +87,38 @@ def file(self, url, download_path, role=None):
             raise NetworkFailure(f"download {description}") from e
 
         return filename
+
+    def _fetch_and_write_content(self, response, filename):
+        """Write the content from the Requests response to file.
+
+        The data is written in to a temporary file in a filesystem location
+        designated by the platform. This avoids partially downloaded files
+        masquerading as complete downloads in later Briefcase runs.
+        The temporary file is only moved to ``filename`` if the download
+        is successful; otherwise, it is deleted.
+
+        :param response: ``requests.Response``
+        :param filename: full filesystem path to save data
+        """
+        try:
+            with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+                total = response.headers.get("content-length")
+                if total is None:
+                    temp_file.write(response.content)
+                else:
+                    progress_bar = self.tools.input.progress_bar()
+                    task_id = progress_bar.add_task("Downloader", total=int(total))
+                    with progress_bar:
+                        for data in response.iter_content(chunk_size=1024 * 1024):
+                            temp_file.write(data)
+                            progress_bar.update(task_id, advance=len(data))
+
+            self.tools.shutil.move(temp_file.name, filename)
```

Comment:
I just realized this `move` may not be a simple inode update if the system's temporary directory is on a filesystem different than that of Briefcase's cache directory. In that case, Python will fall back to a `copy` and `rm`.

Therefore, the underlying problem still exists if a user in this situation sends CTRL+C during the move potentially leaving an incomplete file in the Briefcase cache. (There's also the issue of the `move` potentially taking a relatively long time.)

Given this, I'll pause for feedback:
- Should I move forward with this approach? Or re-implement this solution always creating a temp file within the Briefcase cache?

The primary advantage IMO of the `NamedTemporaryFile` file is if anything goes wrong, the file already exists in an OS-designated temporary directory with (potential) OS faculties to delete it whenever.

I can also wrap the file move in a transient Wait Bar in case it is slow.

Reply:
hmm....another thought is if the Briefcase cache is on a different filesystem, that is likely because the default filesystem is low on space....and that same default filesystem is likely to be used by Python's `NamedTemporaryFile`. Therefore, using it for this purpose here would probably betray some users' configuration....guess I have my answer...

Reply:
on third thought....`NamedTemporaryFile` supports specifying a directory to create the temporary file in....so, that can mitigate the cross-filesystem concerns and still delegate creating a unique filename to Python's `tempfile`. We could even use the support to add a `suffix` or `prefix` to the filename to detect if one of these temporary files persists between Briefcase runs.

Reply:
## tl;dr
Updated the creation of the temporary file to occur in the directory of the Briefcase cache that the file will ultimately be moved to. This mitigates the concerns here